django/db/models/lookups.py

```diff
@@ -282,6 +282,9 @@ def process_rhs(self, compiler, connection):
             return super(In, self).process_rhs(compiler, connection)
 
     def get_rhs_op(self, connection, rhs):
+        if rhs == "()":
```

Comment:
I'm not sure, but maybe we could hack something in the query/queryset itself to not produce (), and instead raise EmptyResultSet directly. That is, somewhere Django is generating the string '()' for Food.objects.all()[0:0], maybe that code could raise EmptyResultSet instead?